Form/ChoosePaymentMethodType.php

```diff
@@ -92,15 +92,18 @@ protected function buildChoiceList(FormBuilderInterface $builder, array $options
             $label = 'form.label.'.$method;
 
             if (Legacy::formChoicesAsValues()) {
-                $options['choices'][$method] = $label;
-            } else {
                 $options['choices'][$label] = $method;
+                if (Legacy::needsChoicesAsValuesOption()) {
+                    $options['choices_as_values'] = true;
```

Comment:
I think it really is correct.
For sf < 2.7 and sf >= 3.0, the option does not exist, and the.
for 2.7 <= sf < 3.0, the option exists, and a deprecation notice is triggered when it's not set to `true`. We should avoid all deprecation notices, which means we have to set it to true, otherwise people using sf < 3.0 will get that deprecation notice **and won't be able to fix it**.